open-telemetry/opentelemetry-collector
# httpprovider for Collector: load configuration from config files in HTTP servers

**Description:** 
An implementation of ConfigMapProvider for HTTP (httpprovider) allows OTEL Collector the ability to load configuration for itself by fetching and reading config files stored in HTTP servers. 

What I changed: added one more new provider implementation folder under confmap/provider/, also added httpprovider.New() in the initialization of a default ConfigProviderSettings.

**Link to tracking Issue:** <Issue number if applicable>
#5810 

**Testing:** 
First, I mocked how a HTTP client works while interacting with a HTTP server.
Second, I designed and passed all the unit tests for edge cases such as empty fields, invalid config files, non-exist config files and so on.
Third, I did integration tests such as:
- Start an OTEL Collector with httpmapprovider.
- Start an OTEL sample application to send telemetry data to check if the OTEL Collector is running well.

**Documentation:** 
Added a README.md

**Sponsor:** 
@Aneurysm9
